sw/device/silicon_creator/lib/drivers/keymgr_functest.c

```diff
@@ -267,6 +244,47 @@ static void soft_reboot(dif_pwrmgr_t *pwrmgr, dif_aon_timer_t *aon_timer) {
   wait_for_interrupt();
 }
 
+/** Key manager configuration steps performed in mask ROM. */
+rom_error_t keymgr_rom_test(void) {
+  ASSERT_OK(keymgr_state_check(kKeymgrStateReset));
+  keymgr_sw_binding_set(&kBindingValueRomExt, &kBindingValueRomExt);
+  keymgr_creator_max_ver_set(kMaxVerRomExt);
+  sec_mmio_check_values(/*rnd_offset=*/0);
+  sec_mmio_check_counters(/*expected_check_count=*/1);
+  return kErrorOk;
+}
+
+/** Key manager configuration steps performed in ROM_EXT. */
+rom_error_t keymgr_rom_ext_test(void) {
+  ASSERT_OK(keymgr_state_check(kKeymgrStateReset));
+
+  const uint16_t kEntropyReseedInterval = 0x1234;
+  ASSERT_OK(keymgr_init(kEntropyReseedInterval));
+
+  sec_mmio_check_values(/*rnd_offset=*/0);
+  keymgr_advance_state();
+  ASSERT_OK(keymgr_state_check(kKeymgrStateInit));
+
+  keymgr_advance_state();
+  ASSERT_OK(keymgr_state_check(kKeymgrStateCreatorRootKey));
+
+  // The software binding register lock is reset after advancing the key
+  // manager, so we need to call this function to update sec_mmio expectation
+  // table.
+  keymgr_sw_binding_unlock_wait();
+  sec_mmio_check_values(/*rnd_offset=*/0);
+
+  keymgr_sw_binding_set(&kBindingValueBl0, &kBindingValueBl0);
+  keymgr_owner_int_max_ver_set(kMaxVerBl0);
+  sec_mmio_check_values(/*rnd_offset=*/0);
+
+  keymgr_advance_state();
+  ASSERT_OK(keymgr_state_check(kKeymgrStateOwnerIntermediateKey));
+
+  sec_mmio_check_counters(/*expected_check_count=*/5);
```

Comment:
q: expected_check_count is 5 because it is also incremented in `sec_mmio_check_counters`, right?

Reply:
Correct. This is to try to catch any faults across checks as well.